assets/src/edit-story/components/autoSaveHandler/index.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useRef, useEffect, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig, useHistory, useStory } from '../../app';
+
+function AutoSaveHandler() {
+  const { autoSaveInterval } = useConfig();
+  const {
+    state: { hasNewChanges },
+  } = useHistory();
+  const {
+    state: {
+      story: { status },
+    },
+    actions: { autoSave, saveStory },
+  } = useStory();
+
+  const [didAutoSave, setDidAutoSave] = useState(false);
+  const timeout = useRef();
+  const isDraft = 'draft' === status;
+
+  // If autoSaveInterval is set to 0 or not defined, don't.
+  if (!autoSaveInterval) {
+    return null;
+  }
+
+  useEffect(() => {
+    // If there are new changes, mark autoSave as not done.
+    if (hasNewChanges && didAutoSave) {
+      setDidAutoSave(false);
+    }
+  }, [hasNewChanges, didAutoSave]);
+
+  useEffect(() => {
+    // @todo This is temporary to ensure only draft gets auto-saved,
+    // until the logic for other statuses has been decided.
+    if (!isDraft) {
+      return;
+    }
+    if (timeout.current) {
+      // Clear the previous timeout if there were any relevant changes meanwhile.
+      clearTimeout(timeout.current);
+    }
+
+    if (!didAutoSave) {
+      timeout.current = setTimeout(() => {
+        if (hasNewChanges) {
+          const update = isDraft ? saveStory : autoSave;
```

Comment:
Let's just always call `autoSave()` here. I think the only path forward here, even when we ready to do auto-save for a published story, would still be to save it into the second draft file.

Reply:
I will remove one of these once it's fully decided about the published story.

However, for a draft, `saveStory` will actually always be called -- since in case of a draft, the original post is used.

So if we're not auto-saving for a published story here for now, `autoSave` would be removed.

Reply:
Got it.